### ISSUE TITLE
Add collapsable answers

### ISSUE THREAD

**Issue by SouthKoreaLN** [stackernews/stacker.news#1422]
**Is your feature request related to a problem? Please describe.**
Nothing important, nothing frustrating. Just thought that if I want to keep posting puzzles like [this one](https://stacker.news/items/696880), it'd be nice to have a way to collapse an answer to hide spoilers. But then again, unlikely this will ever be the primary use of SN, so I won't care much if it does not get implemented :)

**Describe the solution you'd like**
Something like what is illustrated here: https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab

**Describe alternatives you've considered**
Alternative: just ask people not to look at the answers before replying.

**Comment by kravhen** [comment]
Would something like a simple spoiler concealer like this work?
![image](https://github.com/user-attachments/assets/5885a727-968a-4076-adb3-8726957da298)
![sn-spoiler-tag](https://github.com/user-attachments/assets/4ed42165-f26a-470e-8b00-f3d6e18c27e5)

This would use the existing rehype stylers and CSS method for custom tags.

Other options to make it look like a real `<details>`+`<summary>` HTML element as in the gist seems like it'd require a clunkier solution that might not be necessary, like defining a Spoiler component or something.

For the purposes mentioned, the first option seems preferable to me.

Thoughts?

**Comment by SouthKoreaLN** [comment]
This would look like it fits my purpose... as a user, this would just require me adding some tags around the part that needs to be concealed?

**Comment by kravhen** [comment]
Yup, you would write it `<spoiler>`like this`</spoiler>`. Maybe `<hide>` to make it shorter.

**Comment by SouthKoreaLN** [comment]
For me, this would work, if it works for larger blocks of text too.

**Comment by huumn** [comment]
To support more use cases, I'd prefer a more generic approach like github's `details`. 

<details>
    <summary>like this</summary>

This was made with the following markdown:

```md
<details>
    <summary>like this</summary>

This was made with the following markdown:
[...]
</details>
```
</details>

**Comment by SouthKoreaLN** [comment]
Darthcoin gave an interesting suggestion about having to [pay to view the answer](https://stacker.news/items/736461/r/south_korea_ln?commentId=736471). Not fully convinced yet as I was just thinking of adding an extra click before one sees the answer, but in the case this suggestion makes sense, it would be best then to have the sats go to the stacker who posted the hidden answer. This might add a layer of complexity though to this issue.

**Comment by kravhen** [comment]
Should have some time this weekend to give this another try. What bothers me from last time I was digging about this is that it seems to get behavior consistent with github's, with the `details` and `summary` tags and all, it would require `rehype-raw` (+ `rehype-sanitize` for good measure). These are plugins we tried to avoid, and successfully avoided, previously when enabling support for `sub` and `sup`.

An alternative could be to use a custom syntax such as
```
::: spoiler Click to reveal
# Some hidden text
:::
```
The drawback is loss of consistency if we support some tags as expected and familiar in Markdown & HTML, and some other tags using a custom syntax. If going that route, it may become useful to add another widget next to the markdown helper, or a mouseover, that reveals the SN-specific syntax that is supported.

Besides that, I'm wondering if it'd be possible to create a react component to emulate the collapsible section behavior as closely as possible and render that in posts instead. Typing `<details>` could render a Details component rather than HTML, assuming naming it the same as a real tag doesn't cause issue or sanitization.

To be continued

**Comment by huumn** [comment]
It should be able to be done without rehype-raw. I highly recommend asking claude or chatgpt with the context of `rehype-sn`

**Comment by kravhen** [comment]
You are correct! PR coming shortly.
